packages/headless/src/api/search/search-api-client.ts

```diff
@@ -88,11 +95,14 @@ export class SearchAPIClient {
   async search(
     req: SearchRequest
   ): Promise<SearchAPIClientResponse<SearchResponseSuccess>> {
-    const platformResponse = await PlatformClient.call<SearchRequest, Search>({
-      ...baseSearchRequest(req, 'POST', 'application/json', ''),
-      requestParams: pickNonBaseParams(req),
-      renewAccessToken: this.renewAccessToken,
-    });
+    const platformResponse = await PlatformClient.call<SearchRequest, Search>(
+      {
+        ...baseSearchRequest(req, 'POST', 'application/json', ''),
+        requestParams: pickNonBaseParams(req),
+        renewAccessToken: this.renewAccessToken,
+      },
+      this.preprocessRequestMiddleware
```

Comment:
I only added this to the `search` should it also be added to `querySuggest` and the `recommendations` endpoints as well?

Reply:
Yes, I believe so. I see this feature similar to the [requestModifier option](https://docs.coveo.com/en/386/javascript-search-framework/modify-network-calls) on the JSUI. If someone wants to adjust headers, I imagine they want to do this for all endpoints.

Rather than passing the `preprocess` to every endpoint, it would be better to pass it to the platform client when instantiating. Whenever someone invokes `call`, the class would internally call `preprocess` first. It would also be good to add a unit test for the new functionality.

Reply:
`call` currently is a static method so we couldn't pass it into the class without some refactoring. If we make it a true class, and instantiate one inside `SearchApiClient` it either has to be public or it will break our unit tests for `SearchApiClient` since they test that `PlatformClient.call` has been called with the correct values currently.